src/data_structures/fmindex.rs

```diff
@@ -106,44 +79,102 @@ impl FMIndex {
         }
     }
 
-    pub fn occ(&self, r: usize, a: u8) -> usize {
+
+}
+
+/// The Fast Index in Minute space (FM-Index, Ferragina and Manzini, 2000) for finding suffix array
+/// intervals matching a given pattern.
+
+#[cfg_attr(feature = "serde_macros", derive(Serialize, Deserialize))]
+pub struct FMIndex<
+        DBWT: Deref<Target = BWT> + Clone,
+        DLess: Deref<Target = Less> + Clone,
+        DOcc: Deref<Target = Occ> + Clone> {
```

Comment:
Ah, wait, that's right. I tried this before and ran into the issue that `DerefOcc` (for example-- the same goes for the others) defined like `pub type DerefOcc = Deref<Target = Occ>` isn't a trait, but I think I can resolve that via the workaround [here](https://github.com/rust-lang/rust/issues/8634).